ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -1852,120 +1850,128 @@ void create_symbol_table_for_module(ast_node_t* module_items, char * /*module_na
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Input cannot be defined as a reg\n");
+						else if(var_declare->types.variable.is_integer || symbol_table_entry->types.variable.is_integer)
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Input cannot be defined as a integer\n");
+						else if(var_declare->types.variable.is_output || symbol_table_entry->types.variable.is_output )
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Input cannot also be defined as a output\n");
 
-							((ast_node_t*)local_symbol_table_sc->data[sc_spot])->types.variable.is_integer = FALSE;
-							((ast_node_t*)local_symbol_table_sc->data[sc_spot])->types.variable.is_input = FALSE;
+						/* init the entries as inputs */
+						symbol_table_entry->types.variable.is_input = TRUE;
+					}
+					/* MORE ERRORS ... could check for same declaration name ... */
+					else if ( var_declare->types.variable.is_output || symbol_table_entry->types.variable.is_output )
+					{
+						if(var_declare->types.variable.is_input || symbol_table_entry->types.variable.is_input )
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Output cannot also be defined as a input\n");
 
+						/* init the entries as outputs */
+						symbol_table_entry->types.variable.is_output = TRUE;	
+					}
+					
+					if (var_declare->types.variable.is_wire || symbol_table_entry->types.variable.is_wire)
+					{
+						if(var_declare->types.variable.is_reg || symbol_table_entry->types.variable.is_reg)
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Wires cannot be redefined as a reg\n");
+						else if(var_declare->types.variable.is_integer || symbol_table_entry->types.variable.is_integer)
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Wires cannot be redefined as a integer\n");
+						
+						/* init the entries as outputs */
+						symbol_table_entry->types.variable.is_wire = TRUE;
+					}
+					else if(var_declare->types.variable.is_reg || symbol_table_entry->types.variable.is_reg)
+					{
+						if(var_declare->types.variable.is_wire || symbol_table_entry->types.variable.is_wire)
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Reg cannot be redefined as a wire\n");
+						else if(var_declare->types.variable.is_integer || symbol_table_entry->types.variable.is_integer)
+							error_message(NETLIST_ERROR, var_declare->line_number, var_declare->file_number, "%s",
+									"Reg cannot be redefined as a integer\n");
+						
+						/* init the entries as outputs */
+						symbol_table_entry->types.variable.is_reg = TRUE;
+						
+						// TODO: what if we init with a parameter ? I don't think this will work
```

Comment:
with resolve_node added in the init check, it should work - parameters are all resolved at this stage. but that made me think of another problem with symbol table checks thanks